### PR TITLE
Sort Order Implementation for Case Level Grids

### PR DESCRIPTION
**Pull request by anudishCGI**
This PR contains the front end and back end modification of Date of Service Sorting
The date of service from Ambulance, Hospital, Pharmacare, MSP are being implemented to comply with ascending and descending sorting.
Custom label included for sort variables.
Updates functionalities in branch:
Apex Class:
HCCostCaseController
HCCostCaseDAO

LWC Components:
ambulanceRecordsCase
ambulanceRecordsAccount (removing the sort access)
hospitalRecordsCase
hospitalRecordsAccount (removing the sort access)
continuingcareCase  (removing the sort access)
continuingcareAccount  (removing the sort access)
futurecareCase  (removing the sort access)
futurecareAccount  (removing the sort access)
mspRecordsAccount  (removing the sort access)
mspRecordsCase
pharmacareRecordsAccount  (removing the sort access)
pharmacareRecordsCase

I will be raising more PR to accommodate Account level grid changes after this is merged. 

Could you please review and validate the PR @cgijeffolsen  @NataliaNikishina 
Thank You!

### CODE REVIEW

**Review by NataliaNikishina — APPROVED** [review]
@anudishCGI The branch has been successfuly validated. So I am approving the PR. 
cc: @cgijeffolsen Please review the PR. Thank you!

File: TPL/force-app/main/default/lwc/pharmacareRecordsAccount/pharmacareRecordsAccount.js [hunk 1]
```diff
@@ -10,22 +10,19 @@ const COLUMNS = [
         label: 'Case Number',
         fieldName: 'Case_Number__c',
         type: 'text',
-        sortable: true,
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/lwc/pharmacareRecordsAccount/pharmacareRecordsAccount.js:13]: In the data grid layouts, sortable is only available now on Date of Service. This is achieved by removing the sortable attribute on other fields.

File: TPL/force-app/main/default/classes/HCCostCaseController.cls [hunk 2]
```diff
@@ -25,7 +25,7 @@ public with sharing class HCCostCaseController {
     }
 
     @AuraEnabled
-    public static HCCostCaseDetailWrapper getHealthcareCostsAmbulanceForCase(String caseId, String filterValue, Integer pageSize, Integer pageNumber){
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HCCostCaseController.cls:28]: The controlling attribute sortOrder is now passed to the data access process, and the class has two branches as ascending data and descending data have different requirements.

File: TPL/force-app/main/default/classes/HCCostCaseDAO.cls [hunk 3]
```diff
@@ -1,83 +1,187 @@
...
+String dateId = '';
+String query = '';
+String sourceSystemId = null;
+Boolean costValue = false;
+try{
+    if(sortOrder == null || sortOrder == ''){
+        sortOrder = 'asc';
+    }
+
+    if(sortOrder == Label.TPL_Descending){
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HCCostCaseDAO.cls:15]: It was determined that merging these processes under this branch would create more code that leaving them separate due to building complex strings for the many queries for stepping through the dataset.

File: TPL/force-app/main/default/classes/HCCostCaseDAO.cls [hunk 4]
```diff
@@ -1,83 +1,187 @@
...
+    // Populate DateID to maximum value
+        dateId = '99999999zz';
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HCCostCaseDAO.cls:17]: In the case of descending data, a value that exceeds that maximum value is used as the starting location.

**Review by cgijeffolsen — APPROVED** [review]
@anudishCGI Approved

**Pull request merged**